blocks/table-v2/src/app.tsx

```diff
@@ -0,0 +1,304 @@
+        {
+          [columnIdKey]: String(Date.now()),
+          [columnTitleKey]: `Column ${localColumns.length + 1}`,
+        },
+      ],
+    });
+  };
+
+  const addNewRow = async () => {
+    const newRows = [...rows, {}];
+
+    return updateEntity({ [localRowsKey]: newRows });
+  };
+
+  // this should definitely be a sync function
+  const handleCellsEdited: DataEditorProps["onCellsEdited"] = (newValues) => {
+    const newRows = produce(rows, (draftRows) => {
+      for (const { value, location } of newValues) {
+        const [colIndex, rowIndex] = location;
+
+        const columnId = columns[colIndex]?.id;
+
+        if (columnId) {
+          // @ts-ignore
+          draftRows[rowIndex][columnId] = value.data!;
+        }
+      }
+    });
+
+    updateEntity({ [localRowsKey]: newRows });
+
+    return true;
+  };
+
+  const setTitle = async (val: string) => {
+    await updateEntity({ [titleKey]: val });
+  };
+
+  const updateEntity = async (newProperties: RootEntity["properties"]) => {
+    await graphModule?.updateEntity({
+      data: {
+        entityId: blockEntityId,
+        entityTypeId: blockEntityTypeId,
+        properties: { ...blockEntity.properties, ...newProperties },
+      },
+    });
+  };
+
+  const columns: GridColumn[] = localColumns.map((col) => ({
+    id: col[columnIdKey],
+    title: col[columnTitleKey] ?? "",
+    width: 200,
+    hasMenu: !readonly,
+  }));
+
+  const justClickedHeaderRef = useRef(false);
+  const handleHeaderMenuClick = useCallback<
+    NonNullable<DataEditorProps["onHeaderMenuClick"]>
+  >((col, bounds) => {
+    justClickedHeaderRef.current = true;
+    setHeaderMenu({ col, bounds });
+  }, []);
+
+  const { layerProps, renderLayer } = useLayer({
+    isOpen: !!headerMenu,
+    auto: true,
+    placement: "bottom-end",
+    triggerOffset: 2,
+    onOutsideClick: () => {
+      if (justClickedHeaderRef.current) {
+        return (justClickedHeaderRef.current = false);
+      }
+      setHeaderMenu(undefined);
+    },
+    trigger: {
+      getBounds: () => ({
+        left: headerMenu?.bounds.x ?? 0,
+        top: headerMenu?.bounds.y ?? 0,
+        width: headerMenu?.bounds.width ?? 0,
+        height: headerMenu?.bounds.height ?? 0,
+        right: (headerMenu?.bounds.x ?? 0) + (headerMenu?.bounds.width ?? 0),
+        bottom: (headerMenu?.bounds.y ?? 0) + (headerMenu?.bounds.height ?? 0),
+      }),
+    },
+  });
+
+  const selectedRowCount = selection.rows.length;
+
+  const getRowThemeOverride: DataEditorProps["getRowThemeOverride"] = (row) =>
+    row % 2 ? { bgCell: "#f9f9f9" } : undefined;
+
+  return (
+    <div className={styles.block} ref={blockRootRef}>
+      <div className={styles.titleWrapper}>
+        <TableTitle onChange={setTitle} title={title} readonly={readonly} />
+        {!readonly && (
+          <Settings blockEntity={blockEntity} updateEntity={updateEntity} />
+        )}
+      </div>
+      {!!selectedRowCount && !readonly && (
```

Comment:
Once a single checkbox is selected, we should then display checkboxes next to all other rows (in unchecked state, unless/until flipped).

We also want to allow for multiple checkboxes to then be selected, without having to shift+click or cmd+click (if not already supported).

Reply:
Re: displaying all checkboxes

https://github.com/glideapps/glide-data-grid/commit/91ad40b8bc39a22d98994530530b037083509ab9 `glide-data-grid` will add a row-marker option called `checkbox-visible`  on their next patch (`5.2.2`). I suggest waiting until they release this (they updated their repo 9days ago, I guess they’ll release new patch soon)

Reply:
Addressed the other issue here https://github.com/hashintel/hash/pull/1946/commits/13b8a7de657b465ab8f27366cc7887841936dfb6